packages/api/src/providers/__tests__/oauth1-deregister.test.ts

```diff
@@ -70,68 +41,164 @@ describe("oauth1-deregister", () => {
   });
 
   it("deregisters users if no cxid provided", async () => {
-    getUserTokenByUATMock.mockReturnValueOnce([userTokenMocked, anotherUserTokenMocked]);
+    getUserTokenByUATMock.mockReturnValueOnce([userToken_1, userToken_2]);
 
-    saveUserTokenMock.mockReturnValueOnce(updatedUserToken);
-    saveUserTokenMock.mockReturnValueOnce(anotherUserTokenModified);
+    saveUserTokenMock.mockReturnValueOnce(userToken_1_upd);
+    saveUserTokenMock.mockReturnValueOnce(userToken_2_upd);
 
-    updateProviderDataMock.mockReturnValue(testUserModified);
+    updateProviderDataMock.mockReturnValue(testUser_1_upd);
 
-    await testGarmin.deregister([testUser.providerMap.garmin.token]);
+    await testGarmin.deregister([testUser_1.providerMap.garmin.token]);
 
     expect(getUserTokenByUATMock).toHaveBeenCalledTimes(1);
     expect(getUserTokenByUATMock).toHaveBeenCalledWith({
-      oauthUserAccessToken: testUser.providerMap.garmin.token,
+      oauthUserAccessToken: testUser_1.providerMap.garmin.token,
     });
+
     expect(saveUserTokenMock).toHaveBeenCalledTimes(2);
+
+    const firstToken = saveUserTokenMock.mock.calls[0][0];
+    const secondToken = saveUserTokenMock.mock.calls[1][0];
+
+    expect(firstToken).toEqual(
+      expect.objectContaining({
+        oauthUserAccessToken: undefined,
+        oauthUserAccessSecret: undefined,
+      })
+    );
+    expect(secondToken).toEqual(
+      expect.objectContaining({
+        oauthUserAccessToken: undefined,
+        oauthUserAccessSecret: undefined,
+      })
+    );
+
     expect(updateProviderDataMock).toHaveBeenCalledTimes(2);
     expect(updateProviderDataMock).toHaveBeenCalledWith({
-      id: updatedUserToken.userId,
-      cxId: updatedUserToken.cxId,
+      id: userToken_1_upd.userId,
+      cxId: userToken_1_upd.cxId,
       provider: "garmin",
       providerItem: undefined,
     });
     expect(updateProviderDataMock).toHaveBeenCalledWith({
-      id: anotherUserTokenModified.userId,
-      cxId: anotherUserTokenModified.cxId,
+      id: userToken_2_upd.userId,
+      cxId: userToken_2_upd.cxId,
       provider: "garmin",
       providerItem: undefined,
     });
   });
 
   it("deregisters users from the correct cx", async () => {
```

Comment:
Reading the test here its not obvious that `testUser_1.cxId` is the same as `userToken_1_upd.cxId` - which is validated on line 116.

Also, we're not checking the token for the right user was passed to `saveUserToken` - example of why we would check some sort of identification there.